resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+
+## 2. Project Scope
+
+### 3. Use Cases
+
+U1. As a manager, I want to create a new, empty project with a given name.
+
+U2. As a manager, I want to create a new, empty task with a given task name, description, due date, and list of assigned team members.
+
+U3. As a manager, I want to add a task, to a project with a given name.
+
+U4. As a manager, I want to update a task with different information.
+
+U5. As a manager, I want to delete a task, that has been added to a project.
+
+U6. As a manager, I want to view all projects I have created.
+
+U7. As a team member, I want to view all tasks assigned to me.
+
+U8. As a team member, I want to view all tasks assigned to me in a prioritized order (due date).
+
+U9. As a team member, I want to be able to update the progress of specific task (Not Started, In Progress, Complete).
+
+### 4. Stretch Goals
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+U11. Admin can set roles for a user to be a team member or manager. (Multiple roles)
+
+U12. As a team member, I want to be able to comment on a specific task I am assigned to.
+
+U13. As a manager, I want to be able to share and collaborate files and documents with my team.
+
+
+## 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+## 6. API
+
+### 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ProjectModel
+String projectId;
+String projectName;
+String projectStatus;
+List<Task> projectTasks;
+List<UserModel> projectTeamMembers;
+```
+
+```
+// TaskModel
+String taskId;
+String taskName;
+String taskDescription;
+String taskDueDate;
+String taskStatus;
+List<UserModel> taskAssignedUsers;
+Integer taskTimer;
+```
+
+```
+// UserModel
+String userId;
+String userName;
+String userRole;
+```
+
+
+### 6.3. _Create Project Endpoint_
+* Accepts `POST` requests to `/projects`
+* Accepts data to create a new Project with a given name, project status, and an optional list of tasks. Returns the new project, including a unique project ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```"projectName": "Project Name", ```<br>
```

Comment:
If you're going to keep these on their own lines use single backticks instead of triple (the triple is used when you're spanning multiple lines).